File: apps/mobile/ios/lifeagents/SharedDefaults.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(SharedDefaults, NSObject)
RCT_EXTERN_METHOD(getPendingFileName:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(getPendingFileURL:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(clearPendingFileName:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: apps/mobile/ios/lifeagents/SharedDefaults.swift
import Foundation

@objc(SharedDefaults)
class SharedDefaults: NSObject {

    @objc func getPendingFileName(
        _ resolve: @escaping RCTPromiseResolveBlock,
        rejecter reject: @escaping RCTPromiseRejectBlock
    ) {
        let defaults = UserDefaults(suiteName: "group.app.lifeagents.mobile")
        resolve(defaults?.string(forKey: "pendingShareFileName"))
    }

    @objc func getPendingFileURL(
        _ resolve: @escaping RCTPromiseResolveBlock,
        rejecter reject: @escaping RCTPromiseRejectBlock
    ) {
        let defaults = UserDefaults(suiteName: "group.app.lifeagents.mobile")
        resolve(defaults?.string(forKey: "pendingShareFileURL"))
    }

    @objc func clearPendingFileName(
        _ resolve: @escaping RCTPromiseResolveBlock,
        rejecter reject: @escaping RCTPromiseRejectBlock
    ) {
        let defaults = UserDefaults(suiteName: "group.app.lifeagents.mobile")
        defaults?.removeObject(forKey: "pendingShareFileName")
        defaults?.removeObject(forKey: "pendingShareFileURL")
        resolve(nil)
    }

    @objc static func requiresMainQueueSetup() -> Bool { return false }
}
